### ISSUE TITLE
disconnected navmesh

### ISSUE THREAD

**Issue by ghost** [UnvanquishedAssets/map-perseus_src.dpkdir#5]
This map have several disconnected navmeshes:

### Backside of alien base (fixed by https://github.com/UnvanquishedAssets/map-perseus_src.dpkdir/pull/6)

<details> 
  <summary>Details of first problem, now fixed</summary>
   Here, no class can go through (no idea what prevents the navmesh to be generated though, it might be something on the map, or something from the generator):
![2022-01-02-170924_1366x758_scrot](https://user-images.githubusercontent.com/1316300/147881985-f24abc9e-2bea-4048-9bc8-b1111ff16036.png)
This is the most problematic bug, since this makes it very easy to humans to attack undefended alien base. Fixing it would also drastically increase the chances for tyrants to move of alien base.
</details>

### Unconnected top of stairs
Tyrants can't go outside of alien base (this is likely due to the size of margins. Not that tyrants would not be able to leave anyway because there's a bug a bit after on that map with the handling of the stepsize by bots):
![2022-01-02-171205_1366x758_scrot](https://user-images.githubusercontent.com/1316300/147881995-1ce4bca3-a656-4dd5-8981-7f299dfe4c56.png)

### Some passage over the void

Small aliens and humans don't know how to reach the other side. (Tyrants are not affected, though, and human players do not need to jump to reach the other size, momentum is enough to avoid falling in the hole. I'd even say, it's almost impossible to use that hole, making it of dubious interest, a bit like some yocto's).
![2022-01-02-171239_1366x758_scrot](https://user-images.githubusercontent.com/1316300/147882001-84475272-bac6-4895-982b-533870d64f6d.png)

Note that all of those problems can be worked around simply by placing navcons.

**Comment by ghost** [comment]
@illwieckz , @slipher : do you think adding navcons to maps would be acceptable fixes? Would they work inside a .dpk?
If yes, I can provide fixes for the most problematic things I'm aware of.

**Comment by necessarily-equal** [comment]
I updated this issue's description to make it more clear what there is still to fix

**Comment by necessarily-equal** [comment]
I think navcons could be a good workaround for now

**Comment by illwieckz** [comment]
Perseus from Unvanquished 0.54.0 is shipped with navcons from https://github.com/sweet235/navcon

If those navcons had links for those parts then it's fixed.

**Comment by necessarily-equal** [comment]
I haven't seen one for the tyrant step at least

**Comment by ghost** [comment]
IMO using navmeshes for the place where there is a hole is an acceptable workaround to close this problem.
Area near the entrance though are affected by a bug (misconfigured feature) in the navmesh generation code. I guess it would be ok to move this issue to another tracker though, as this is not solely a mapping error.
To describe the bug in few words, the problem is that recast generates margins to avoid agents to fall or to go too close to walls. This feature generates the same kind of problems as discussed here on few other non-official maps, and does not even prevents bots to fall off the cliff in chasm.
Here, the problem is that this passage requires a tyrant to force it's way against geometry, that is, tyrants are (slightly) bigger than the passage, and I **believe** (I have not investigated this particular situation, this is just guessing) the margins prevent recast to connect the areas despite the fact it is possible to navigate.

**Comment by ghost** [comment]
> IMO using navmeshes for the place where there is a hole is an acceptable workaround to close this problem.

Obviously, I mean offMesh connections, aka navcon in unvanquished's parler.

**Comment by ghost** [comment]
Remaining problems have been fixed by autojump, except for the void passage, which needs a navcon.

**Comment by bmorel** [comment]
²

**Comment by ghost** [comment]
marauders (not advanced), mantis, dretch, and humans are still affected by the "void" problem. This can be fixed by navcons only.
